client/my-sites/checkout/composite-checkout/hooks/use-create-payment-complete-callback.tsx

```diff
@@ -3,6 +3,7 @@ import { resolveDeviceTypeByViewPort } from '@automattic/viewport';
 import { isURL } from '@wordpress/url';
 import debugFactory from 'debug';
 import { useTranslate } from 'i18n-calypso';
+import { flatten } from 'lodash';
```

Comment:
Would [Array.prototype.flat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) work as well here? It would let us avoid adding a lodash dependency (p4TIVU-9Bf-p2)

Reply:
I don't think using `flat` will work. We have the following structure in the object. 

```javascript
const transaction = {
  "receiptId": "3124324",
  "purchases": {
    "21321321": [ // <- Site ID
       {
          ...
          "saas_redirect_url": "url1"
       },
       {
          ...
          "saas_redirect_url": "url2"
       }
    ]
  }
};
```
We want to go through each site's purchase and look for the `saas_redirect_url` field. So having flat on purchases object won't work. Also, the `flat` is an array method and does not work on the `object`. The simplest thing I can do is retrieve the object keys and loop through the array in a nested loop. 

Anyways, based on @/sirbrillig suggestions above, We need to append `saas_redirect_url` to get the receipt endpoint instead of appending `saas_redirect_url ` in the transactions endpoint.